components/server/src/code-sync/code-sync-service.ts

```diff
@@ -4,303 +4,303 @@
-            if (!manifest.latest.extensions) {
-                manifest.latest.extensions = fromTheiaRev;
-            }
-            if (!manifest.latest.settings) {
-                manifest.latest.settings = fromTheiaRev;
-            }
-            res.json(manifest);
-            return;
-        });
-        router.get('/v1/resource/:resource', async (req, res) => {
-            if (!User.is(req.user)) {
-                res.sendStatus(204);
-                return;
-            }
-            const resourceKey = ALL_SERVER_RESOURCES.find(key => key === req.params.resource);
-            const revs = resourceKey && await this.db.getResources(req.user.id, resourceKey)
-            if (!revs || !revs.length) {
-                res.sendStatus(204);
-                return;
-            }
-            const result: { url: string, created: number }[] = revs.map(e => ({
-                url: req.originalUrl + '/' + e.rev,
-                created: Date.parse(e.created) / 1000 /* client expects in secondsm */
-            }));
-            res.json(result);
-            return;
-        });
-        router.get('/v1/resource/:resource/:ref', async (req, res) => {
-            if (!User.is(req.user)) {
-                res.sendStatus(204);
-                return;
-            }
-            const resourceKey = ALL_SERVER_RESOURCES.find(key => key === req.params.resource);
-            if (!resourceKey) {
-                res.sendStatus(204);
-                return;
-            }
-            let resourceRev: string | undefined = req.params.ref;
-            if (resourceRev !== fromTheiaRev) {
-                resourceRev = (await this.db.getResource(req.user.id, resourceKey, resourceRev))?.rev;
-            }
-            if (!resourceRev && (resourceKey === SyncResource.Extensions || resourceKey === SyncResource.Settings)) {
-                resourceRev = fromTheiaRev;
-            }
-            if (!resourceRev) {
-                res.setHeader('etag', '0');
-                res.sendStatus(204);
-                return;
-            }
-            if (req.headers['If-None-Match'] === resourceRev) {
-                res.sendStatus(304);
-                return;
-            }
+             if (!isWithFunctionAccessGuard(req) || !req.functionGuard?.canAccess(accessCodeSyncStorage)) {
+                 res.sendStatus(403);
+                 return;
+             }
+             return next();
+         });
+         router.get('/v1/manifest', async (req, res) => {
+             if (!User.is(req.user)) {
+                 res.sendStatus(204);
+                 return;
+             }
+             const manifest = await this.db.getManifest(req.user.id);
+             if (manifest?.latest && !manifest.latest.extensions) {
+                 manifest.latest.extensions = fromTheiaRev;
+             }
+             if (manifest?.latest && !manifest.latest.settings) {
+                 manifest.latest.settings = fromTheiaRev;
+             }
```

Comment:
Maybe this can be removed safely? no need to keep returning new entries with `fromTheiaRev`

Reply:
I think you are right, no reason to keep it since it is never going to be triggered anyways. Added it.